data/data/bootstrap/files/usr/local/bin/installer-gather.sh

```diff
@@ -71,53 +79,79 @@ function queue() {
+queue resources/namespaces.json oc --request-timeout=5s get namespaces -o json
+queue resources/nodes.json oc --request-timeout=5s get nodes -o json
+queue resources/openshiftapiserver.json oc --request-timeout=5s get openshiftapiserver -o json
+queue resources/pods.json oc --request-timeout=5s get pods --all-namespaces -o json
+queue resources/rolebindings.json oc --request-timeout=5s get rolebindings --all-namespaces -o json
+queue resources/roles.json oc --request-timeout=5s get roles --all-namespaces -o json
 # this just lists names and number of keys
-queue resources/secrets-names.txt oc --kubeconfig=/opt/openshift/auth/kubeconfig --request-timeout=5s get secrets --all-namespaces
+queue resources/secrets-names.txt oc --request-timeout=5s get secrets --all-namespaces
 # this adds annotations, but strips out the SA tokens and dockercfg secrets which are noisy and may contain secrets in the annotations
-queue resources/secrets-names-with-annotations.txt oc --kubeconfig=/opt/openshift/auth/kubeconfig --request-timeout=5s get secrets --all-namespaces -o=custom-columns=NAMESPACE:.metadata.namespace,NAME:.metadata.name,TYPE:.type,ANNOTATIONS:.metadata.annotations | grep -v -- '-token-' | grep -v -- '-dockercfg-'
-queue resources/services.json oc --kubeconfig=/opt/openshift/auth/kubeconfig --request-timeout=5s get services --all-namespaces -o json
+queue resources/secrets-names-with-annotations.txt oc --request-timeout=5s get secrets --all-namespaces -o=custom-columns=NAMESPACE:.metadata.namespace,NAME:.metadata.name,TYPE:.type,ANNOTATIONS:.metadata.annotations | grep -v -- '-token-' | grep -v -- '-dockercfg-'
+queue resources/services.json oc --request-timeout=5s get services --all-namespaces -o json
 
-FILTER=gzip queue resources/openapi.json.gz oc --kubeconfig=/opt/openshift/auth/kubeconfig --request-timeout=5s get --raw /openapi/v2
+FILTER=gzip queue resources/openapi.json.gz oc --request-timeout=5s get --raw /openapi/v2
 
 echo "Waiting for logs ..."
 wait
 
+# The existence of the file located in LOG_BUNDLE_BOOTSTRAP_ARCHIVE_NAME is used
+# as indication that a bootstrap process has already previously taken place and the resulting
+# bundle already exists in the filesystem. In that case, we replace some of the logs we've
+# just collected with the logs from said bundle.
+#
+# An example for a scenario where this happens is when we're running inside a single-node
+# bootstrap-in-place deployment post-pivot master node, rather than a typical bootstrap node.
+# In that case, the bootstrap node collects and bundles logs pre-reboot and transfers that bundle
+# via an ignition file to this post-pivot master node.
+LOG_BUNDLE_BOOTSTRAP_NAME="log-bundle-bootstrap"
+LOG_BUNDLE_BOOTSTRAP_ARCHIVE_NAME="/var/log/${LOG_BUNDLE_BOOTSTRAP_NAME}.tar.gz"
+
+if [[ -f ${LOG_BUNDLE_BOOTSTRAP_ARCHIVE_NAME} ]]; then
+    echo "Including existing bootstrap bundle ${LOG_BUNDLE_BOOTSTRAP_ARCHIVE_NAME}"
+    tar -xzf ${LOG_BUNDLE_BOOTSTRAP_ARCHIVE_NAME}
```

Comment:
> Do we need to extract the archive or can we just add the archive as is to the artifacts?
Choosing one over the other is rather arbitrary. I lean towards extracting it because it's more convenient for the user. Also "double compression" i.e. `.tar.gz` inside `.tar.gz` feels wrong.

> What is the working directory where these extracted files are going to be left?
`~`
I'll change it so it gets extracted directly into the bundle instead

Reply:
Sounds good. I prefer extracting it, too.

Reply:
Fixed broken link, the diff is actually https://github.com/eranco74/installer/pull/19

Reply:
Fixed